src/tweenable.rs

```diff
@@ -894,264 +917,399 @@ mod tests {
+
+        let mut transform = Transform::default();
+        let mut prev_times_completed = 0;
+        for (
+            delta,
+            (progress, times_completed, event_count, direction, expected_state, expected_transform),
+        ) in expected_values
+        {
+            // Tick the tween
+            let actual_state = tween.tick(
+                delta,
+                &mut transform,
+                dummy_entity,
+                &mut event_writer.get_mut(&mut world),
+            );
+
+            // Propagate events
+            {
+                let mut events = world.get_resource_mut::<Events<TweenCompleted>>().unwrap();
+                events.update();
+            }
+
+            // Check actual values
+            assert_eq!(direction, tween.direction());
+            assert_eq!(expected_state, actual_state);
+            assert!(
+                abs_diff_eq(tween.progress(), progress, 1e-5),
+                "progress: expected={progress}, actual={}",
+                tween.progress()
+            );
+            assert_eq!(times_completed, tween.times_completed(), "times_completed");
+            assert!(
+                transform
                     .translation
-                    .abs_diff_eq(expected_translation, 1e-5));
-                assert!(transform.rotation.abs_diff_eq(Quat::IDENTITY, 1e-5));
+                    .abs_diff_eq(expected_transform.translation, 1e-5),
+                "translation: expected={}, actual={}",
+                expected_transform.translation,
+                transform.translation
+            );
+            assert!(
+                transform
+                    .rotation
+                    .abs_diff_eq(expected_transform.rotation, 1e-5),
+                "rotation: expected={}, actual={}",
+                expected_transform.rotation,
+                transform.rotation
+            );
+            assert!(
+                transform.scale.abs_diff_eq(expected_transform.scale, 1e-5),
+                "scale: expected={}, actual={}",
+                expected_transform.scale,
+                transform.scale
+            );
+            let cb_mon = callback_monitor.lock().unwrap();
+            assert_eq!(
+                event_count as u64, cb_mon.invoke_count,
+                "cb_mon.invoke_count"
+            );
+            assert_eq!(
+                times_completed, cb_mon.last_reported_count,
+                "cb_mon.last_reported_count"
+            );
 
-                // Clear callback
-                tween.clear_completed();
-                assert!(tween.on_completed.is_none());
+            {
+                let just_completed = prev_times_completed != times_completed;
+                prev_times_completed = times_completed;
+
+                let mut event_reader = event_reader.get_mut(&mut world);
+                if just_completed {
+                    assert!(!event_reader.is_empty());
+                    for event in event_reader.iter() {
```

Comment:
Should we assert there's a single event here? Since the docs got changed to state that.

Reply:
Good point, done.